codice/ddf
# DDF-4362: Adds configurable, hyphenated uuid generation in web ui

#### What does this PR do?
Adds hyphenated uuid generation in web ui (for queries); adds support to configure enabling/disabling hyphenated property in admin console.

Admin console code largely contributed by @bellcc 
#### Who is reviewing it? 

@bellcc 
@Corey-Collins 
@brianfelix 
@jrnorth 
@glenhein 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@bdeining
#### How should this be tested?
1. Navigate to admin console -> `System`
2. Find `Metacard UUID Generator Service` and check/uncheck `Use Hyphens` property (attached screenshots)
3. Using search ui create a workspace
4. Observe queries in the workspace (using browser console, etc.) should contain (non)hyphenated ids based on the selection in 2.
#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4362](https://codice.atlassian.net/browse/DDF-4362)
#### Screenshots
<img width="1792" alt="screen shot 2018-11-28 at 10 01 46 am" src="https://user-images.githubusercontent.com/118054/49160754-79d96600-f2f5-11e8-9e95-8c7741c1dc1e.png">
<img width="1792" alt="screen shot 2018-11-28 at 10 01 54 am" src="https://user-images.githubusercontent.com/118054/49160755-79d96600-f2f5-11e8-9d0f-f5af01836732.png">


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
